src/models/device.ts

```diff
@@ -630,7 +612,9 @@ const getDeviceModel = function (
 			if (device == null) {
 				throw new errors.BalenaDeviceNotFound(uuidOrId);
 			}
-			return device as Device;
+			return device as NonNullable<
```

Comment:
I think it would make sense to not have this cast (which can hide issues) and instead change the above to:
```ts
			let device; NonNullable<
				OptionsToResponse<Device['Read'], T, typeof uuidOrId>
			>;
```
so that attempts to assign something that doesn't match to `device` will fail at the root cause

Reply:
Well I think we can go with 
```
let device: OptionsToResponse<Device['Read'], T, typeof uuidOrId> | OptionsToResponse<Device['Read'], T, undefined>[number];

```
as for the particular case here these are not mutually exchangeable